Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp

```diff
@@ -251,10 +262,18 @@ void AuthenticatorCoordinator::discoverFromExternalSource(const Document& docume
     }
 
     if (requestOptions.signal) {
-        requestOptions.signal->addAlgorithm([weakThis = WeakPtr { *this }](JSC::JSValue) {
+        m_isCancelling = true;
+        requestOptions.signal->addAlgorithm([weakThis = WeakPtr { *this }](JSC::JSValue) mutable {
             if (!weakThis)
                 return;
-            weakThis->m_client->cancel();
+            weakThis->m_client->cancel([weakThis = WTFMove(weakThis)] () mutable {
+                if (!weakThis)
+                    return;
+                if (auto queuedRequest = WTFMove(weakThis->m_queuedRequest)) {
+                    weakThis->m_isCancelling = false;
```

Comment:
This and the line below should be indented at same level (maybe they are and the GitHub UI is confusing.)

Reply:
Indented at the same level as the lambda initiation? Or both ifs equal to each other (the latter is already true).

Reply:
ditto

Reply:
ditto